tests_peerscout_api/unit_tests/main_test.py

```diff
@@ -174,6 +184,11 @@ def _get_ok_json(response):
     return response.json
 
 
+class TestGetTargetDatasetEnv:
+    def test_should_read_target_dataset_as_ci_for_e2e_test(self):
```

Comment:
I believe this is not directly related to end to end tests. Ideally the unit test wouldn't depend on environment variables set outside..

In Data Hub Core DAGs there is a `mock_env` fixture defined: https://github.com/elifesciences/data-hub-core-airflow-dags/blob/develop/tests/unit_test/conftest.py#L22-L25

You could copy it into `conftest.py` in this repo too and use it here like:

```python
@pytest.mark.usefixtures('mock_env')
def test_should_return_ci_by_default(self):
    assert get_target_dataset_env() == 'ci'
    # this could also refer to DEFAULT_PEERSCOUT_API_TARGET_DATASET_VALUE (in which case the test name would be changed slightly)


def test_should_return_value_of_peerscout_api_target_dataset_env_var(self, mock_env: dict):
    mock_env['PEERSCOUT_API_TARGET_DATASET'] = 'dataset1'
    assert get_target_dataset_env() == 'dataset1'

```